.circleci/config.yml

```diff
@@ -3,52 +3,60 @@ orbs:
   anchore: anchore/anchore-engine@1.4.0
 jobs:
   Full Build:
-    machine:
+    machine: &virtual_machine
       image: circleci/classic:latest
       docker_layer_caching: true
     steps:
-      - checkout
-      - run:
-          name: Install java 11
+      - run: &install_java
+          name: Install Java 11
           command: |
             sudo add-apt-repository ppa:openjdk-r/ppa
-            sudo apt-get update -q
-            sudo apt install -y openjdk-11-jdk
-      - run:
-          name: Set default java 11
-          command:
-            sudo update-java-alternatives --set java-1.11.0-openjdk-amd64
-      - run:
-          name: Building
-          command: |
-            ./gradlew clean build --info
+            sudo apt-get -qq update
+            sudo apt-get -qq install -y openjdk-11-jdk
+      - run: &configure_java
+          name: Change to Java 11
+          command: sudo update-java-alternatives --set java-1.11.0-openjdk-amd64
+      - checkout
+      - restore_cache: &restore_cache
```

Comment:
✏️ Is `restore_cache` and `save_cache` needed if everything is happening in one job?